src/coreclr/System.Private.CoreLib/src/System/Reflection/TypeNameParser.CoreCLR.cs

```diff
@@ -80,6 +84,54 @@ namespace System.Reflection
             }.Parse();
         }
 
+        internal static RuntimeType GetTypeByNameUsingCARules(string typeName, RuntimeModule scope)
+        {
+            ArgumentException.ThrowIfNullOrEmpty(typeName);
+
+            RuntimeAssembly requestingAssembly = scope.GetRuntimeAssembly();
+
+            RuntimeType? type = (RuntimeType?)new TypeNameParser(typeName)
+            {
+                _throwOnError = true,
+                _suppressContextualReflectionContext = true,
+                _requestingAssembly = requestingAssembly
```

Comment:
```suggestion
                _requestingAssembly = requestingAssembly,
                _suppressCASearchRules = false
```

I'm recommending the above only because the concept of "CA search rules" is incredibly vague and it helps indicate what to look for in this method.

Reply:
CASearchRules are gone from the method and field names.